app/uk/gov/hmrc/transitmovements/repositories/DeparturesRepository.scala

```diff
@@ -215,4 +218,40 @@ class DeparturesRepositoryImpl @Inject() (
     })
 
   }
+
+  def getDeparture(departureId: DepartureId): EitherT[Future, MongoError, Option[Departure]] =
+    mongoRetry(Try(collection.find(Filters.eq("_id", departureId.value))) match {
+      case Success(obs) =>
+        obs.headOption().map {
+          opt => Right(opt)
+        }
+      case Failure(NonFatal(ex)) =>
+        Future.successful(Left(UnexpectedError(Some(ex))))
+    })
+
+  def updateMessages(departureId: DepartureId, message: Message): EitherT[Future, MongoError, Unit] =
+    getDeparture(departureId)
+      .map {
+        departureOpt =>
+          departureOpt.map(
+            _.copy(updated = OffsetDateTime.ofInstant(clock.instant, ZoneOffset.UTC), messages = NonEmptyList(message, departureOpt.get.messages.toList))
+          )
+      }
+      .map {
+        updatedDeparture =>
+          mongoRetry(Try(collection.replaceOne(Filters.eq("_id", departureId.value), updatedDeparture.get)) match {
+            case Success(obs) =>
+              obs.toFuture().map {
+                result =>
+                  if (result.wasAcknowledged()) {
+                    Right(())
+                  } else {
+                    Left(UpdateNotAcknowledged(s"Message update failed for departure"))
+                  }
+              }
+            case Failure(NonFatal(ex)) =>
+              Future.successful(Left(UnexpectedError(Some(ex))))
+          })
+      }
+
```

Comment:
Am I missing something or does this not account for a Departure/Movement that cannot be found? I was just checking for integrating with my ticket and I can't see how I'd ever recieve a 404